### PR TITLE
use insert to change _id

### LINKED ISSUE #63: Mod on _id not allowed
When trying to use cloudpipe connected with ObjectRocket's Mongo, this error popped up:

```
cloudpipe_1 | ERRO[0347] Authentication failure.                       error=Unable to communicate with storage: exception: Mod on _id not allowed
```

### PR DESCRIPTION
**Pull request by jrperritt**
This should fix #63

### CODE REVIEW

**Comment by jrperritt** [comment]
Er, looks like Travis got hung up with GoDeps on go tip.

**Comment by rgbkrk** [comment]
Looking good to me, I'll check out Travis now.

**Comment by rgbkrk** [comment]
That is such a useless error from godep

```
godep: restore: signal: killed
```

File: storage.go [hunk 1]
```diff
@@ -218,14 +218,17 @@ func (storage *MongoStorage) UpdateJob(job *SubmittedJob) error {
...
+	err := storage.accounts().FindId(name).One(&acct)
+	if err != nil && err != mgo.ErrNotFound {
 		return nil, err
 	}
+	if err == mgo.ErrNotFound {
+		err = storage.accounts().Insert(out)
+		if err != nil {
+			return nil, err
+		}
+	}
```
Comment (smashwilson) [inline storage.go]: Dumb question - doesn't this introduce a race condition? It seems to me that it's possible another service could create an account with the same name in between the `FindId` call and the `Insert` call.

If the upsert isn't an option with Mongo 2.6, maybe we could check the `err` returned by `Insert` and re-query if it's indicating a duplicate `_id`?

Comment (jrperritt) [inline storage.go]: Hmm, possibly. Are there multiple services trying to create this document? If that's the case, (I *think*) we could wrap it in a [transaction](http://godoc.org/labix.org/v2/mgo/txn#Op).

Comment (smashwilson) [inline storage.go]: The idea is that we can scale this service horizontally - run a ton of cloudpipe containers and load-balance among them. I could easily see two requests coming in from the same user being routed to two instances of this service and having them fight.

A transaction would be :+1: too.

**Comment by jrperritt** [comment]
Can't auth right now to test the most recent commit, but I _think_ it'll work.

**Pull request closed without merging**